bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/ReadHandle.java

```diff
@@ -111,4 +112,32 @@
      */
     long getLength();
 
+    /**
+     * The type contains LAC and a LedgerEntry want to read.
+     * It is used for readLastAddConfirmedAndEntry.
+     */
+    @Data
+    class LastConfirmedAndEntry {
+        private final Long lac;
+        private final LedgerEntry entry;
+    }
+
+    /**
+     * Asynchronous read specific entry and the latest last add confirmed.
+     * If the next entryId is less than known last add confirmed, the call will read next entry directly.
+     * If the next entryId is ahead of known last add confirmed, the call will issue a long poll read
+     * to wait for the next entry <i>entryId</i>.
+     *
+     * @param entryId
+     *          next entry id to read
+     * @param timeOutInMillis
```

Comment:
in case of THIS timeout the CompletableFuture will return null ? or is will be completed exceptionally ?

Reply:
it will return `null` entry and latest lac on timeout.

Reply:
Yes. In LedgerHandle, most of the similar situation will return null.

Reply:
Can we add a line which explains? That was my real request, sorry

Reply:
Can we add a line which explains? That was my real request, sorry

Reply:
Thanks. sure will add it.